client/my-sites/stats/controller.jsx

```diff
@@ -167,7 +167,34 @@ export function redirectToDefaultModulePage( context ) {
 }
 
 export function insights( context, next ) {
-	context.primary = <StatsInsights />;
```

Comment:
It might be better to omit the `startDate` query string functionality from this PR. Two reasons:

1. `startDate` query string is unintuitive for the user.
    - `startDate` only controls the annual highlights section, but its name implies it may affect the whole page.
    - `startDate` stores a full ISO date (e.g., `startDate=2016-12-16`), even though we only use the year.
2. The increased technical complexity of this controller is hard to justify for the benefit of being able to hard-link the annual highlights section state. 
    - Note that we'll need to migrate `moment` usage in this controller to `luxon` if we keep this controller logic. (I understand this logic is probably copied from the traffic controller.)

I understand doing so will require replacing our use of the `StatsPeriodNavigation` component with something more lightweight (maybe some simple Buttons with Icons inside?).

Reply:
Ok, I will simplify it. I took this approach because I thought a solution unified with other pages would be the right choice. However, I agree with your logic, and it looks like the query string is a bit too much at the moment.

Reply:
[The new approach](https://github.com/Automattic/wp-calypso/pull/71343) drastically simplified the whole component, even with new navigation arrows.